mmv1/products/vertexai/IndexEndpointDeployedIndex.yaml

```diff
@@ -0,0 +1,256 @@
+      - !ruby/object:Api::Type::Integer
+        name: 'minReplicaCount'
+        description: The minimum number of replicas this DeployedModel will be always deployed on.
+        default_from_api: true
+      - !ruby/object:Api::Type::Integer
+        name: 'maxReplicaCount'
+        description: The maximum number of replicas this DeployedModel may be deployed on when the traffic against it increases. If maxReplicaCount is not set, the default value is minReplicaCount
+        default_from_api: true
+  - !ruby/object:Api::Type::NestedObject
+    name: dedicatedResources
+    description: A description of resources that are dedicated to the DeployedIndex, and that need a higher degree of manual configuration.
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: 'machineSpec'
+        description: The minimum number of replicas this DeployedModel will be always deployed on.
+        immutable: true
+        required: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'machineType'
+            description: The type of the machine.
+            immutable: true
+      - !ruby/object:Api::Type::Integer
+        name: 'minReplicaCount'
+        description: The minimum number of machine replicas this DeployedModel will be always deployed on. This value must be greater than or equal to 1.
+        required: true
+      - !ruby/object:Api::Type::Integer
+        name: 'maxReplicaCount'
+        description: The maximum number of replicas this DeployedModel may be deployed on when the traffic against it increases. If maxReplicaCount is not set, the default value is minReplicaCount
+        default_from_api: true
+  - !ruby/object:Api::Type::Boolean
+    name: enableAccessLogging
+    description: If true, private endpoint's access logs are sent to Cloud Logging.
+    immutable: true
+    default_value: false
+  - !ruby/object:Api::Type::NestedObject
+    name: deployedIndexAuthConfig
+    description: If set, the authentication is enabled for the private endpoint.
+    immutable: true
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: authProvider
+        description: Defines the authentication provider that the DeployedIndex uses.
+        properties:
+          - !ruby/object:Api::Type::Array
+            name: audiences
+            description: The list of JWT audiences. that are allowed to access. A JWT containing any of these audiences will be accepted.
+            immutable: true
+            item_type: Api::Type::String
+          - !ruby/object:Api::Type::Array
+            name: allowedIssuers
+            immutable: true
+            description: |
+              A list of allowed JWT issuers. Each entry must be a valid Google service account, in the following format: service-account-name@project-id.iam.gserviceaccount.com
+            item_type: Api::Type::String
+  - !ruby/object:Api::Type::Array
+    name: reservedIpRanges
+    immutable: true
```

Comment:
You could put a comment here about how the API accepts new values without error but doesn't process the update  to the fields correctly. It'll be easier for people to read that versus reading through your PR (with all the automation messages to scroll through!)

Reply:
(I'd want us to resolve Slack convo before committing to this being immutable)

Reply:
since we confirmed that this would be immutable the extra comment wouldn't be necessary correct?

Reply:
I think it'd be useful to provide context that the API accepts changed values but doesn't use them, versus returning an error saying that the field cannot be updated. That's the information that a newcomer wouldn't be able to learn from the API docs

Reply:
TBH it'll also help with reviewing this PR. Comparing the schema to the API docs is a lot of work so it'd be useful to have a comment in the schema that explains why it doesn't match the API docs

Reply:
Resolved by https://github.com/GoogleCloudPlatform/magic-modules/pull/11039#discussion_r1707602322